File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution(object):
    def numberOfSubarrays(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        cnt=0
        dicta={}
        dicta[0]=1
        res=0
        for i in range(len(nums)):
            if nums[i]%2!=0:
                cnt+=1
            
            if cnt-k in dicta:
                res+=dicta.get(cnt-k)
            dicta[cnt]=dicta.get(cnt,0)+1
        return res
      

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
int func(vector<int>& nums, int k){
    int n=nums.size();
    int cnt=0, ans=0;
    int i=0;
    for(int j=0;j<n;j++){
        if(nums[j]%2!=0)
          cnt++;
        while(cnt>k){
            if(nums[i]%2!=0)
                cnt--;
            i++;
            }
            ans+=(j-i+1);
        }
        return ans;
}
    int numberOfSubarrays(vector<int>& nums, int k) {
    return func(nums,k)-func(nums,k-1);   
    }
};